src/node_buffer.cc

```diff
@@ -1223,6 +1223,18 @@ static void EncodeInto(const FunctionCallbackInfo<Value>& args) {
   results[1] = written;
 }
 
+static void IsUtf8(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsTypedArray() || args[0]->IsArrayBuffer());
+  ArrayBufferViewContents<char> abv(args[0]);
+
+  if (abv.WasDetached()) {
+    return node::THROW_ERR_BUFFER_CONTEXT_NOT_AVAILABLE(env->isolate());
```

Comment:
I don't know which error is the right one – it should probably be similar to what you get for `new Uint8Array(buffer)` when `buffer` is detached – but this is _definitely_ the wrong one.

Reply:
`new Uint8Array(buffer)` throws a very specific error. I don't know what this function should throw. Any recommendations?

```
TypeError: Cannot perform Construct on a detached ArrayBuffer
    at new Uint8Array (<anonymous>)
```